packages/core/src/lib/waku.ts

```diff
@@ -68,28 +65,28 @@ export class WakuNode implements Waku {
   constructor(
     options: WakuOptions,
     libp2p: Libp2p,
-    store?: (components: StoreComponents) => IStore,
-    lightPush?: (components: LightPushComponents) => ILightPush,
-    filter?: (components: FilterComponents) => IFilter,
+    store?: (libp2p: Libp2p) => IStore,
+    lightPush?: (libp2p: Libp2p) => ILightPush,
+    filter?: (libp2p: Libp2p) => IFilter,
     peerExchange?: (components: PeerExchangeComponents) => IPeerExchange
   ) {
     this.libp2p = libp2p;
 
-    const { peerStore, connectionManager, registrar } = libp2p;
-    const components = { peerStore, connectionManager, registrar };
-
     if (store) {
-      this.store = store(components);
+      this.store = store(libp2p);
     }
     if (filter) {
-      this.filter = filter(components);
+      this.filter = filter(libp2p);
     }
     if (lightPush) {
-      this.lightPush = lightPush(components);
+      this.lightPush = lightPush(libp2p);
     }
 
     if (peerExchange) {
-      this.peerExchange = peerExchange(components);
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore: Libp2p is now hiding internal components but peer discovery
```

Comment:
@danisharora099 if we can remove peer exchange from waku here, then we cna remove this ts ignore.